From 96c01c88363d6e5e9b7519bc4e8b5d86cf783e1f Mon Sep 17 00:00:00 2001
From: Jan Tojnar <jtojnar@gmail.com>
Date: Sat, 16 Mar 2019 05:46:20 +0100
Subject: [PATCH] Port to gupnp-1.2

gupnp_context_get_host_ip has been deprecated for a long time, as
the host-ip property moved to gssdp.

There is also a ton of new deprecations but I will address that separately.
---
 configure.ac            | 8 ++++----
 libdleyna/server/upnp.c | 4 ++--
 test/dbus/dms-info.c    | 2 --
 3 files changed, 6 insertions(+), 8 deletions(-)
 mode change 100755 => 100644 libdleyna/server/upnp.c
 mode change 100755 => 100644 test/dbus/dms-info.c

diff --git a/configure.ac b/configure.ac
index 11879e7..61f4fca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,10 +36,10 @@ LT_LANG([C])
 
 # Checks for libraries.
 PKG_PROG_PKG_CONFIG(0.16)
-PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.28])
-PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.28])
-PKG_CHECK_MODULES([GSSDP], [gssdp-1.0 >= 0.13.2])
-PKG_CHECK_MODULES([GUPNP], [gupnp-1.0 >= 0.20.3])
+PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.36])
+PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.36])
+PKG_CHECK_MODULES([GSSDP], [gssdp-1.2 >= 0.13.2])
+PKG_CHECK_MODULES([GUPNP], [gupnp-1.2 >= 0.20.3])
 PKG_CHECK_MODULES([GUPNPAV], [gupnp-av-1.0 >= 0.11.5])
 PKG_CHECK_MODULES([GUPNPDLNA], [gupnp-dlna-2.0 >= 0.9.4])
 PKG_CHECK_MODULES([SOUP], [libsoup-2.4 >= 2.28.2])
diff --git a/libdleyna/server/upnp.c b/libdleyna/server/upnp.c
old mode 100755
new mode 100644
index 9913a76..d04efb8
--- a/libdleyna/server/upnp.c
+++ b/libdleyna/server/upnp.c
@@ -351,8 +351,8 @@ static void prv_device_unavailable_cb(GUPnPControlPoint *cp,
 
 	udn = gupnp_device_info_get_udn((GUPnPDeviceInfo *)proxy);
 
-	ip_address = gupnp_context_get_host_ip(
-		gupnp_control_point_get_context(cp));
+	ip_address = gssdp_client_get_host_ip(
+		GSSDP_CLIENT(gupnp_control_point_get_context(cp)));
 
 	if (!udn || !ip_address)
 		goto on_error;
diff --git a/test/dbus/dms-info.c b/test/dbus/dms-info.c
old mode 100755
new mode 100644
index a189b9f..4530e58
--- a/test/dbus/dms-info.c
+++ b/test/dbus/dms-info.c
@@ -360,8 +360,6 @@ int main(int argc, char *argv[])
 	if (sigprocmask(SIG_BLOCK, &mask, NULL) == -1)
 		goto on_error;
 
-	g_type_init();
-
 	/* Create proxy for com.intel.dLeynaServer.Manager.  The Manager
 	object has no properties.  We will create the proxy asynchronously
 	and use G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES to ensure that
-- 
2.20.1

